metadata-ingestion/src/datahub/ingestion/source/sql/vertica.py

```diff
@@ -537,49 +629,83 @@ def _process_projections(
-        try:
-            # SQLAlchemy stubs are incomplete and missing this method.
-            # PR: https://github.com/dropbox/sqlalchemy-stubs/pull/223.
-            projection_info: dict = inspector.get_projection_comment(projection, schema)  # type: ignore
-        except NotImplementedError:
-            return description, properties, location
-        except ProgrammingError as error:
+        from datahub.ingestion.source.ge_data_profiler import GEProfilerRequest
+
+        tables_seen: Set[str] = set()
+        profile_candidates = None  # Default value if profile candidates not available.
+        yield from super().loop_profiler_requests(inspector, schema, sql_config)
+        for projection in inspector.get_projection_names(schema):
+            dataset_name = self.get_identifier(
+                schema=schema, entity=projection, inspector=inspector
+            )
+
+            if not self.is_dataset_eligible_for_profiling(
+                dataset_name, sql_config, inspector, profile_candidates
+            ):
+                if self.config.profiling.report_dropped_profiles:
+                    self.report.report_dropped(f"profile of {dataset_name}")
+                continue
+            if dataset_name not in tables_seen:
+                tables_seen.add(dataset_name)
+            else:
+                logger.debug(f"{dataset_name} has already been seen, skipping...")
+                continue
+            missing_column_info_warn = self.report.warnings.get(MISSING_COLUMN_INFO)
+            if (
+                missing_column_info_warn is not None
+                and dataset_name in missing_column_info_warn
+            ):
+                continue
+            (partition, custom_sql) = self.generate_partition_profiler_query(
+                schema, projection, self.config.profiling.partition_datetime
+            )
+            if partition is None and self.is_table_partitioned(
+                database=None, schema=schema, table=projection
+            ):
+                self.report.report_warning(
+                    "profile skipped as partitioned table is empty or partition id was invalid",
+                    dataset_name,
+                )
+                continue
+            if (
+                partition is not None
+                and not self.config.profiling.partition_profiling_enabled
+            ):
+                logger.debug(
+                    f"{dataset_name} and partition {partition} is skipped because profiling.partition_profiling_enabled property is disabled"
+                )
+                continue
+            self.report.report_entity_profiled(dataset_name)
             logger.debug(
-                f"Encountered ProgrammingError. Retrying with quoted schema name for schema {schema} and table {properties} %s",
-                error,
+                f"Preparing profiling request for {schema}, {projection}, {partition}"
+            )
+
+            yield GEProfilerRequest(
```

Comment:
what changed in here vs the superclass? please add it as a comment, and also consider if we can avoid the duplicate code

Reply:
added comments, main changes are we added one more loop in the function which gets projections and done profiling for projection. all logics and other things are same .